File: Digital Business Enablement/bluegather/src/main/java/br/com/fiap/bluegather/repository/AvaliacaoRepository.java
package br.com.fiap.bluegather.repository;

import br.com.fiap.bluegather.dto.AvaliacaoResumoDTO;
import br.com.fiap.bluegather.model.Avaliacao;

import java.util.Set;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

@Repository
public interface AvaliacaoRepository extends JpaRepository<Avaliacao, Long> {
    Set<Avaliacao> findByEventoId(Long eventoId);

    @Query("SELECT new br.com.fiap.bluegather.dto.AvaliacaoResumoDTO(a.evento.id, COUNT(a), AVG(a.nota)) FROM Avaliacao a WHERE a.evento.id = :eventoId GROUP BY a.evento.id")
    AvaliacaoResumoDTO findAvaliacaoResumoByEventoId(@Param("eventoId") Long eventoId);

    Set<Avaliacao> findByAvaliadorId(Long avaliadorId);
}

File: Enterprise Application Development/bluegather/Repositories/AvaliacaoRepository.cs
using bluegather.Dtos;
using BlueGather.Context;
using BlueGather.Dto;
using BlueGather.Models;
using Microsoft.EntityFrameworkCore;
using System.Linq;
using System.Threading.Tasks;

namespace BlueGather.Repositories
{
    public class AvaliacaoRepository : Repository<AvaliacaoModel>
    {
        public AvaliacaoRepository(BlueGatherContext blueGatherContext) : base(blueGatherContext) { }

        public async Task<AvaliacaoResumoDto> FindAvaliacaoResumoByEventoId(long eventoId)
        {
            return await _blueGatherContext.Avaliacao
                .Where(a => a.Evento.Id == eventoId)
                .GroupBy(a => a.Evento.Id)
                .Select(g => new AvaliacaoResumoDto
                {
                    IdEvento = g.Key,
                    QtdAvaliadores = g.LongCount(),
                    MediaNota = g.Average(a => a.Nota)
                })
                .FirstOrDefaultAsync();
        }
    }
}